platform/engineering/design-docs/separate-content-build.md

```diff
@@ -0,0 +1,477 @@
+**no simple way to sync** that change.
+
+In order for the application build to create a landing page consistent with the
+rest of VA.gov, it would need access to the templates. This is the kind of
+dependency entanglement this document aims to resolve.
+
+##### Types of drift
+
+###### Missing application pages
+If we had to revert a content release for whatever reason, we may lose an
+application page.
+
+**Mitigation:** With hundreds of content editors, it's not feasible to revert a
+content release. Our current solution is to deploy a new release with the fix.
+This approach minimizes the risk of this type of drift.
+
+###### Contents of an application landing page change
+If the contents of a landing page change to, for example, point to a different
+bundle name before that bundle was made available, the request for it would 404
+in production.
+
+**Mitigation:** That said, the contents of the pages shouldn't change between
+releases unless we did it deliberately. In that case, it's the responsibility of
+the author of the change to ensure the change is valid before deploying it.
+
+###### Rolling back a newly-released application's code, but not landing page
+If we release a new application and had to roll the application deployment back,
+we could end up with a landing page that points to a JS bundle that doesn't
+exist.
+
+**Mitigation:** Similar to [missing application
+pages](#missing-application-pages), our approach has historically been to
+release a fix instead of reverting a release.
+
+###### Landing page deployed before code
+The content release which includes a newly prod-facing application landing page
+is deployed before the code is deployed to production.
+
+**Mitigation:** It's up to the team releasing a new application to ensure the
+code is in production before deploying the landing page for it.
+
+
+### Work Estimates
+The following estimates vary greatly depending on who's doing the work.
+
+1. Write a standalone script that can be pointed to the build output directory
+   to check for broken links
+    - **Estimate:** 2 hours - 2 days
+1. Trigger this job in Jenkins (Nomad? Circle?) after a `master` build succeeds
+    - **Estimate:** 1 - 2 days
+1. Remove the broken link checker step from the Metalsmith script
+    - **Estimate:** < 1 hour
+1. Remove Webpack from the Metalsmith build script
+   `vets-website`
+    - Use
+      [`html-webpack-plugin`](https://github.com/jantimon/html-webpack-plugin)
+      to generate temporary application landing pages
+      - Create a template which we can use with the above plugin
+      - This should only be used in `localhost` builds until `vets-website` is
+        no longer responsible for deploying the content
+    - Write a script to run content and application builds while they're both in
+      `vets-website`
+    - **Estimate:** 1 - 3 days
+1. Modify the `createReactPages` step to use [the new JSON
```

Comment:
does the CMS team need to sign off on this approach?

Reply:
This part is essentially the same as it has been. The part where we'll want alignment is the bit where we add the React landing pages as pages in the CMS. I evidently haven't outlined that bit very well, so once I do, I'll ask somebody from that team to check it out. :+1:

Reply:
I added more about this in the `React application landing pages` section.